### PR TITLE
extra log files: remove legacy

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Closes https://github.com/cylc/cylc-flow/issues/5610
* Remove residual code leftover from the obsolete `extra log files` runtime configuration option.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/data_messages.proto [hunk 1]
```diff
@@ -144,7 +144,6 @@ message PbJob {
     optional float execution_time_limit = 14;
     optional string platform = 15;
     optional string job_log_dir = 17;
-    repeated string extra_logs = 29;
```
Comment (oliver-sanders) [inline cylc/flow/data_messages.proto:147]: @dwsutherland 

This field is no longer needed, is it safe to remove a "repeated string"?

Could this cause compatibility problems between cylc-flow and cylc-uiserver at different versions?

**Review by MetRonnie — APPROVED** [review]

File: cylc/flow/data_messages_pb2.py [hunk 2]
```diff
@@ -34,68 +33,68 @@
...
-  _globals['_TDELTAS']._serialized_start=8122
-  _globals['_TDELTAS']._serialized_end=8297
-  _globals['_TPDELTAS']._serialized_start=8300
-  _globals['_TPDELTAS']._serialized_end=8486
-  _globals['_WDELTAS']._serialized_start=8489
-  _globals['_WDELTAS']._serialized_end=8684
-  _globals['_ALLDELTAS']._serialized_start=8687
-  _globals['_ALLDELTAS']._serialized_end=8896
+  _PBMETA._serialized_start=34
+  _PBMETA._serialized_end=184
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/data_messages_pb2.py:37]: Makes sense, I'll regenerate this based on the outcome of the above comment.

File: cylc/flow/data_messages.proto (hunk 1 above)
Comment (dwsutherland) [reply cylc/flow/data_messages.proto:147]: I think this is fine, since I don't think we ask for this field anywhere.. And as long as we don't reuse `29`..

**Review by dwsutherland — APPROVED** [review]
LGTM 👍

**Comment by dwsutherland** [comment]
> Actually after looking at the docs a little, it seems we need to reserve the deleted field number: https://protobuf.dev/programming-guides/proto3/#deleting

As long as the UIS/UI/TUI never specifically asked for the field, I think we're fine.
The reason for for reserving the field is to not break clients (UIS) that were specifically asking for that field (which we just need to confirm we weren't (which I don't think we are https://github.com/cylc/cylc-flow/blob/5b76dbaa3d3acc0e3bb11810e9501fb5c2d2fdc0/cylc/flow/data_store_mgr.py#L289).

File: cylc/flow/data_messages_pb2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dwsutherland) [reply cylc/flow/data_messages_pb2.py:37]: I think it's safe to use within and up to the minor version, i.e. with `pip freeze` version:
```
protobuf==4.21.12

```
I used `protoc` version:
```
sutherlander@cortex:~$ protoc --version

libprotoc 3.21.12

```

[the remark this answers is not in the document]
Comment (dwsutherland) [reply cylc/flow/data_messages_pb2.py:37]: https://github.com/protocolbuffers/protobuf/releases/tag/v21.12

[the remark this answers is not in the document]
Comment (dwsutherland) [reply cylc/flow/data_messages_pb2.py:37]: We should probably upgrade at some point lol .. might put up a PR after a few of these (including the no-rewalk) has gone in.

**Comment by dwsutherland** [comment]
Tested it,
- UIS this branch (regenerated the `data_messages_pb2.py`
- Scheduler master

and found working:
image

File: cylc/flow/data_messages_pb2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply cylc/flow/data_messages_pb2.py:37]: Can upgrade whenever, the only reason we pin this dependency so hard is that we haven't found out what the protobuf compatibility policy is yet, i.e. the span of versions should be inter-compatible.

**Comment by oliver-sanders** [comment]
Right so, conclusions:

* Compiling with an older protobuf version is fine and possibly safer.
* Removing the field is ok so long as we were never using it.

**Comment by oliver-sanders** [comment]
I've tested this with the UIS/Scheduler running this-branch/8.2.1 and vice versa, it seems to be working fine.

**Comment by MetRonnie** [comment]
Thinking this through, I would imagine we still wouldn't want to be in the position where:
1. We've added a new field with this same field number
2. A new UIS asks for this field number from an old scheduler, thus getting the old `extra_log_files` result instead of whatever it was actually asking for

File: cylc/flow/data_messages_pb2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MetRonnie) [reply cylc/flow/data_messages_pb2.py:37]: Frustrating lack of info on the compatibility of protoc - https://github.com/protocolbuffers/protobuf/issues/11123

**Comment by oliver-sanders** [comment]
Enough to jam a comment in to "reserve" the old field number?

**Comment by MetRonnie** [comment]
I think all that's needed is
```diff
 message PbJob {
+  reserved 29;
   optional string stamp = 1;
   optional string id = 2;
```
https://protobuf.dev/programming-guides/proto3/#deleting,

**Comment by MetRonnie** [comment]
Did you regenerate the `pb2` file after reserving the field?

**Comment by oliver-sanders** [comment]
Yes: https://github.com/cylc/cylc-flow/compare/abf2d031a4e0bbef1b8744bbe88c9e5bf2ee84ae..840096b2009e0c179e933b14218843f5ce17207e

**Review by MetRonnie — APPROVED** [review]
Testing with mixed versions of scheduler/UIS didn't bring up any problems

**Pull request merged**